pkg/ee/kubevirt-network-controller/controller.go

```diff
@@ -256,5 +256,10 @@ func cleanUpKubevirtCloudProviderNetworkPolicy(ctx context.Context, kvInfraClien
 			Namespace: dc.NamespacedMode.Namespace,
 		},
 	}
-	return kvInfraClient.Delete(ctx, networkPolicy)
+	if err := kvInfraClient.Delete(ctx, networkPolicy); err != nil {
+		if apierrors.IsNotFound(err) {
+			return err
+		}
+	}
+	return nil
```

Comment:
This looks wrong, I think you want `cleanUpKubevirtCLoudProviderNetworkPolicy` to return `nil` when the `NetworkPolicy` cannot be found (= it's already been deleted, which is good).

```suggestion
	if err := kvInfraClient.Delete(ctx, networkPolicy); err != nil && !apierrors.IsNotFound(err) {
		return err
	}
	return nil
```

Reply:
There's also ctrlruntimeclient.IgnoreNotFound(err) for these situations.

Reply:
^ that is the even better suggestion.

Reply:
Thank you @embik for the quick review. I changed that as suggested by you. Forgot the exclamation mark.